src/diagnostics/land_compute_methods.jl

```diff
@@ -0,0 +1,145 @@
+    return quote
+        $(functions...)
+    end
+end
+
+### BucketModel ###
+
+# variables stored in p (diagnostics variables stored in the cache)
+@land_compute "aerodynamic_resistance" BucketModel p.bucket.turbulent_fluxes.r_ae
+@land_compute "albedo" BucketModel p.bucket.α_sfc
+@land_compute "latent_heat_flux" BucketModel p.bucket.turbulent_fluxes.lhf
+@land_compute "net_radiation" BucketModel p.bucket.R_n
+@land_compute "sensible_heat_flux" BucketModel p.bucket.turbulent_fluxes.shf
+@land_compute "surface_air_density" BucketModel p.bucket.ρ_sfc
+@land_compute "surface_specific_humidity" BucketModel p.bucket.q_sfc
+@land_compute "surface_temperature" BucketModel p.bucket.T_sfc
+@land_compute "vapor_flux" BucketModel p.bucket.turbulent_fluxes.vapor_flux
+
+# variables stored in Y (prognostic or state variables)
+@land_compute "snow_water_equivalent" BucketModel Y.bucket.σS
+@land_compute "soil_temperature" BucketModel Y.bucket.T
+@land_compute "subsurface_water_storage" BucketModel Y.bucket.W
+@land_compute "surface_water_content" BucketModel Y.bucket.Ws
+
+### SoilCanopyModel ###
+
+# variables stored in p (diagnostics variables stored in the cache)
+
+## Canopy Module ##
+
+# Canopy - Solar Induced Fluorescence
+@land_compute "solar_induced_fluorescence" SoilCanopyModel p.canopy.sif.SIF
+
+# Canopy - Autotrophic respiration
+@land_compute "autotrophic_respiration" SoilCanopyModel p.canopy.autotrophic_respiration.Ra
+
+# Canopy - Conductance
+@land_compute "stomatal_conductance" SoilCanopyModel p.canopy.conductance.gs
+@land_compute "canopy_transpiration" SoilCanopyModel p.canopy.conductance.transpiration
+
+# Canopy - Energy
+@land_compute "canopy_aerodynamic_resistance" SoilCanopyModel p.canopy.energy.r_ae
+@land_compute "canopy_latent_heat_flux" SoilCanopyModel p.canopy.energy.lhf
+@land_compute "canopy_sensible_heat_flux" SoilCanopyModel p.canopy.energy.shf
+
+# Canopy - Hydraulics
+@land_compute "canopy_water_potential" SoilCanopyModel p.canopy.hydraulics.ψ
+@land_compute "cross_section" SoilCanopyModel p.canopy.hydraulics.fa
+@land_compute "cross_section_roots" SoilCanopyModel p.canopy.hydraulics.fa_roots
+@land_compute "leaf_area_index" SoilCanopyModel p.canopy.hydraulics.area_index.leaf
+@land_compute "moisture_stress_factor" SoilCanopyModel p.canopy.hydraulics.β
+@land_compute "root_area_index" SoilCanopyModel p.canopy.hydraulics.area_index.root
+@land_compute "stem_area_index" SoilCanopyModel p.canopy.hydraulics.area_index.stem
+
+# Canopy - Photosynthesis
+@land_compute "photosynthesis_net_canopy" SoilCanopyModel p.canopy.photosynthesis.GPP
+@land_compute "photosynthesis_net_leaf" SoilCanopyModel p.canopy.photosynthesis.An
+@land_compute "respiration_leaf" SoilCanopyModel p.canopy.photosynthesis.Rd
+@land_compute "vcmax25" SoilCanopyModel p.canopy.photosynthesis.Vcmax25
+
+# Canopy - Radiative Transfer
+@land_compute "near_infrared_radiation" SoilCanopyModel p.canopy.radiative_transfer.inc_nir
```

Comment:
Maybe this could be called down as well? Like SW_down?

Reply:
NIR_down you mean - will do, I agree.